src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -2917,11 +2961,21 @@ static PaError MakeWaveFormatFromParams(WAVEFORMATEXTENSIBLE *wavex, const PaStr
         useExtensible = TRUE;
     }
 
+    if ((streamInfo != NULL) && (streamInfo->flags & paWinWasapiAc3Passthrough))
+    {
+        isAc3Passthrough = TRUE;
```

Comment:
Probably you should check whether `streamInfo->flags & paWinWasapiUseChannelMask` is set (?and return an error if it is not. Unfortunately I don't know what error to return). Otherwise, if `paWinWasapiUseChannelMask` is not set by the caller, channelMask will be zero.

Alternatively you could copy the channelMask here even if `paWinWasapiUseChannelMask` is not set.

Reply:
> Alternatively you could copy the channelMask here even if paWinWasapiUseChannelMask is not set.

You mean I assign a default value to `channelMask`? Such as those preset in the ac3 and eac3 test code.